static/js/click_handlers.js

```diff
@@ -413,6 +429,36 @@ exports.initialize = function () {
         $(".tooltip").remove();
     });
 
+    // Tooltips for left sidebar icons.
+    $("#streams_filter_icon").on("mouseenter", function (e) {
+        e.stopPropagation();
+
+        const elem = $(e.currentTarget);
+        const title_text = i18n.t("Filter streams");
+        const title = title_text + " (q)";
+
+        ui.create_generic_tooltip(elem, title, "left");
+    });
+
+    $("#streams_filter_icon").on("mouseleave", function (e) {
+        e.stopPropagation();
+        $(e.currentTarget).tooltip('hide');
+    });
+
+    $("#streams_inline_cog").on("mouseenter", function (e) {
+        e.stopPropagation();
```

Comment:
I think I see a nice chance to move the event handlers inside the `create_generic_tooltip` function. There are at least 3 instances in this PR alone where we add `mouseenter` and then `mouseleave` events in a very similar manner.